server/server.go

```diff
@@ -258,6 +281,7 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 	handleFunc("/callback/{connector}", s.handleConnectorCallback)
 	handleFunc("/approval", s.handleApproval)
 	handleFunc("/healthz", s.handleHealth)
+	handleFunc("/metrics", promhttp.HandlerFor(c.PrometheusRegistry, promhttp.HandlerOpts{}).ServeHTTP)
```

Comment:
Is it okay that we're serving these at a public endpoint? Request to dex are unauthenticated. Should it be a separate listener?

Reply:
Good point. These metrics may be security sensitive, I'd say we at least put them on a separate listener, then we can segment by network, which should be enough I'd say.

Reply:
Works for me